abdiff/core/calc_ab_metrics.py

```diff
@@ -66,32 +64,31 @@ def create_record_diff_matrix_dataset(
     for i, batch in enumerate(
         diffs_ds.to_batches(
             batch_size=batch_size,
-            columns=["timdex_record_id", "source", "ab_diff", "has_diff"],
+            columns=["timdex_record_id", "source", "modified_timdex_fields", "has_diff"],
         )
     ):
         start_time = time.time()
         batch_df = batch.to_pandas()
 
-        # parse diff JSON to dictionary for batch
-        batch_df["ab_diff"] = batch_df["ab_diff"].apply(
-            lambda diff_json: json.loads(diff_json)
-        )
-
         batch_metrics = []
         for _, row in batch_df.iterrows():
             record_metrics = {
                 "timdex_record_id": row["timdex_record_id"],
                 "source": row["source"],
-                "has_diff": 1 if row["has_diff"] == "true" else 0,
+                "has_diff": (1 if row["has_diff"] == "true" else 0),
             }
-            diff_data = row["ab_diff"]
-            record_metrics.update(generate_field_diff_bools_for_record(diff_data))
+
+            # for each modified field (root key in diff), set column and value = 1 (True)
+            if row["modified_timdex_fields"] is not None:
+                for field in row["modified_timdex_fields"]:
+                    record_metrics[field] = 1
+
```

Comment:
This reads a lot cleaner, `deepdiff` seems like a great change

Reply:
100% agree!  The fact it picks out the modified "root" keys, works perfectly with our TIMDEX records.